### DISCUSSION TITLE
Not showing graph and not saving to log

### DISCUSSION THREAD

**Discussion by RCompassi** [Daves-Astrophotography/UniUSBSQMServer#14] [Q&A] [answered]
Hello!

Thanks for this very cool software!

But I have one issue (using current version 1.3.0).

The graph is not showing and the memory is getting nothing. I can read from the SQM, the software displays the last averaged reading, temperature, etc, but nothing is plotted and nothing is saved.

https://i.postimg.cc/kGxfRM9Z/image.png

APT and NINA gets the data correctly from the SQM Server.

I have enabled file logging.

Any ideas?

Thanks

**Comment by DaveAuld** [comment]
Hi there,

Sorry you are having problems.

I suspect there is nothing drawing due to there being no Raw Value being detected, I can see this in your image the first value Latest Raw Reading is empty.

Can you try a couple of things so I can see what's going on.
1) In the Configuration Logging, increase the poll interval to the fastest 10s.
2) Once this is done and connected to the SQM, leave for 30 seconds or so, i.e. long enough to get a few polls of data then;
- Click on the Serial Poll box on the status bar and click the View Latest Messages
- Click on the Send Message and select Send Command  - ux Unaveraged Data
- Click on the Memory Status box and click the View Memory

Send me some screenshots of what is displayed on your system, and before / after you send the UX command, do you see the latest message update with any difference before and after on the U response

I am wondering if it is maybe an old firmware of your unit etc. If you look at the unit information on my screenshot vs your screenshot, you can see the Protocol / Model / Feature Version are vastly different.

Thanks,
Dave

**Reply by RCompassi** [reply]
After sending the command UX unaveraged data, the graph showed up and started to poll.

Very nice!

**Reply by DaveAuld** [reply]
Still intrigued as to why it didn't work the first time, when the connect is done, the first thing the software does is send the ix, ux and rx commands...... Glad it is working for you now. Do let me know if you continue to see any issues.

**Comment by DaveAuld** [comment]
Further to my post above, I dug through my code and it will only draw data on the graph when it sees a new DataRecord being added to the DataStore. The code that creates a DataRecord, is looking for both the UX and RX (raw and averaged) data responses from the SQM unit before creating a record. 

I'm still leaning towards firmware versions at this point, until you can feedback on the UX response test.

**Comment by DaveAuld** [comment, marked as answer]
Manually triggering a UX command appeared to kick start the updating, underlying reasons for it not functioning as designed initially is not understood as the software does the ix, rx and ux commands on initial connect, then does the ux and rx polls at required interval.

First occasion this behaviour has been seen, so will monitor for future occurrences before changing any code as may have been a local 'glitch'.

Sending the Manual Command via the Serial Poll status bar context menu as a workaround on the occasion resolved.
